VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -612,8 +635,32 @@ Function Set-EsxiCertificateMode {
         $certModeSetting = Get-AdvancedSetting "vpxd.certmgmt.mode" -Entity $vCenterServer.connection
         if ($certModeSetting.value -ne $mode) {
             Set-AdvancedSetting $certModeSetting -Value $mode -confirm:$false
-            Write-Output "Certificate Management Mode is set to $mode on the vCenter Server instance $($vCenterServer.details.fqdn)."
-            Write-Output "Please restart the vCenter Server services for the change to take effect. See the vCenter Server Configuration documentation for information about restarting services."
+            # Restart "VMware Certificate Authority Service" and "VMware Certificate Management Service"
+            Write-Output 'Restarting vCenter Server services("VMware Certificate Authority Service" and "VMware Certificate Management Service") for the change to take effect.'
+            $services = @("certificateauthority", "certificatemanagement")
+            $failedServices = @()
+
+            foreach ($service in $services) {
+                $serviceStatus = Restart-VcenterService $service
+                if ($serviceStatus.status -eq "STOPPED") {
+                    $failedServices += $serviceStatus.name
+                }
+            }
+
+            if ($failedServices.Count -gt 0) {
+                $failedServicesList = [string]::Join(", ", $failedServices)
+                Write-Error "The following services failed to start: $failedServicesList. Operation Failed." -ErrorAction Stop
+            }
+
+            Invoke-RestartService -Service "certificateauthority"
```

Comment:
You will need to change this to call the new function - Restart-VcenterService

Reply:
address in 30a32aca1c5451e5a0829cfb001061f9b16d0042